compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -1078,11 +1078,14 @@ warning.usage.of.deprecated.function=\
 error.object.outside.methods.not.allowed=\
   outside object method definitions are not allowed
 
+error.cannot.infer.type=\
+  cannot infer type here
```

Comment:
Can't we be more specific in this error message? Like give the invokable type signature for which the compiler failed to infer the type? Also shall we rephrase this to something like `failed to infer type` or `failed to infer type of arrow expression`?

Reply:
This can fail for record literals and array literals as well. Let's say we have a function like this in `lang.string` module

```
@TypeParam type T1 string;
@TypeParam type T2 anydata[];

function foo(T1 srt, T2 v) returns [T1,T2];

// other package
string ss = "abc";
var x = ss.foo([]); // Here we can't infer the type.
```

Ref to the error message: I used the same pattern where we start with "cannot" word for error messages. Yes, I think we can improve messages better.